index.js

```diff
@@ -35,30 +35,56 @@ module.exports = class SubscriptionDedupe {
   subscriptions: SubscriptionMap;
   */
 
-  subscribe(topic/*: string */) {
+  subscribe(topic /*: string */) {
     let existing = this.subscriptions[topic];
 
     if (!existing) {
       // Nothing found, create the subscription object
       existing = this.subscriptions[topic] = {
         promise: this.options.onSubscribe(topic),
-        refCount: 0
+        refCount: 0,
+        closing: null,
       };
     }
 
+    if (existing.closing) {
+      existing.closing.isReopened = true;
+      existing.closing = null;
```

Comment:
Won't this overwrite L51?

Reply:
This is necessary to allow multiple pending reopens. The object is still part of the scope of the function deleting the subscription object in L80.